src/devices/vsmart.ts

```diff
@@ -0,0 +1,526 @@
+            throw new Error(`Invalid ${periodName} LED intensity. Expected: 0-${MAX_PERCENTAGE}%`);
+        }
+
+        const intensity = Math.round((percentage * 255) / MAX_PERCENTAGE);
+
+        const payload = {data: [intensity, endpointIndex, 0]};
+
+        await entity.command<"vsmartSwitchControl", "setLedIntensity", VSmartLedControl>("vsmartSwitchControl", "setLedIntensity", payload, {
+            manufacturerCode: VSMART_MANUFACTURER_CODE,
+            disableDefaultResponse: true,
+        });
+
+        return {
+            state: {
+                [key]: percentage,
+            },
+        };
+    },
+});
+
+const vsmartExtend = {
+    customCluster: (): ModernExtend =>
+        m.deviceAddCustomCluster("vsmartSwitchControl", {
+            ID: 0x0000,
+            manufacturerCode: VSMART_MANUFACTURER_CODE,
+            attributes: {},
+            commands: {
+                setLedColor: {
+                    ID: 0xf3,
+                    parameters: [{name: "data", type: Zcl.BuffaloZclDataType.LIST_UINT8}],
+                },
+                setVibrationIntensity: {
+                    ID: 0xf0,
+                    parameters: [{name: "data", type: Zcl.BuffaloZclDataType.LIST_UINT8}],
+                },
+                setLedBrightnessLevel: {
+                    ID: 0xf4,
+                    parameters: [{name: "data", type: Zcl.BuffaloZclDataType.LIST_UINT8}],
+                },
+                setLedIntensity: {
+                    ID: 0xf2,
+                    parameters: [{name: "data", type: Zcl.BuffaloZclDataType.LIST_UINT8}],
+                },
+                setTimePeriod: {
+                    ID: 0xf5,
+                    parameters: [{name: "data", type: Zcl.BuffaloZclDataType.LIST_UINT8}],
+                },
+            },
+            commandsResponse: {
+                setLedColor: {
+                    ID: 0xf3,
+                    parameters: [{name: "data", type: Zcl.BuffaloZclDataType.LIST_UINT8}],
+                },
+                setVibrationIntensity: {
+                    ID: 0xf0,
+                    parameters: [{name: "data", type: Zcl.BuffaloZclDataType.LIST_UINT8}],
+                },
+                setLedBrightnessLevel: {
+                    ID: 0xf4,
+                    parameters: [{name: "data", type: Zcl.BuffaloZclDataType.LIST_UINT8}],
+                },
+                setLedIntensity: {
+                    ID: 0xf2,
+                    parameters: [{name: "data", type: Zcl.BuffaloZclDataType.LIST_UINT8}],
+                },
+                setTimePeriod: {
+                    ID: 0xf5,
+                    parameters: [{name: "data", type: Zcl.BuffaloZclDataType.LIST_UINT8}],
+                },
+            },
+        }),
+    addLedColorControl: (): ModernExtend => {
+        const exposes_list = [
+            exposes
+                .text("ledIndicatorColorOn", ea.SET)
```

Comment:
```suggestion
                .text("led_indicated_color_on", ea.SET)
```

(and do the same for all other exposes)